### PR TITLE
Refs [TASK][36031] Thiết kế giao diện tìm kiếm

### PR DESCRIPTION
**Pull request by khangvv158**
## Related Tickets
- [#Task 36031: Thiết kế giao diện tìm kiếm](https://edu-redmine.sun-asterisk.vn/issues/36031)
## WHAT
+ Danh sách kết quả tìm kiếm :
- Hiển thị hình ảnh phim
- Tên phim
## Evidence (Screenshot or Video)
Screenshot_2021-05-14-15-05-10-649_com sun moviedb_54

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/moviedb_54/screen/search/SearchFragment.kt [hunk 1]
```diff
@@ -30,11 +33,40 @@ class SearchFragment : Fragment() {
...
-        registerLiveData()
+        initViews()
+        handleEvents()
     }
 
-    private fun registerLiveData() {
-        viewModel.movies.observe(viewLifecycleOwner, Observer {
+    private fun initViews() {
+        searchAdapter = SearchAdapter {
+        }
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/moviedb_54/screen/search/SearchFragment.kt]: code gì đây thanh niên?

File: app/src/main/java/com/sun/moviedb_54/ultis/BindingAdapters.kt [hunk 2]
```diff
@@ -11,7 +11,9 @@ import com.sun.moviedb_54.extensions.setVoteRating
 
 @BindingAdapter("bind:onLoadImage")
 fun ImageView.loadImage(url: String?) {
-    loadFromUrl(Constant.BASE_URL_IMAGE + url)
+    if (url?.isNotBlank() == true) {
+        loadFromUrl(Constant.BASE_URL_IMAGE + url)
+    }
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/moviedb_54/ultis/BindingAdapters.kt]: dùng let

File: app/src/main/res/layout/fragment_search.xml [hunk 3]
```diff
@@ -1,18 +1,79 @@
...
+            app:layout_constraintWidth_percent="0.8" />
+
+        <androidx.appcompat.widget.AppCompatTextView
+            android:id="@+id/appCompatTextView"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/dp_20"
+            android:text="@string/results"
+            android:textColor="@color/white"
+            android:textSize="@dimen/sp_30"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_search.xml]: dùng _sdp cho đồng bộ với team. Tương tự các chổ còn lại

File: app/src/main/java/com/sun/moviedb_54/screen/search/adapter/SearchAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return if (getItem(position).id == null) VIEW_TYPE_LOADING else VIEW_TYPE_ITEM
+    }
+
+    override fun setData(data: MutableList<MovieResult>?) {
+        data?.let {
+            submitList(it.toMutableList())
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/moviedb_54/screen/search/adapter/SearchAdapter.kt]:
```suggestion
            submitList(it)
```

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**